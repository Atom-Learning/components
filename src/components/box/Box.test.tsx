import { render, screen } from '@testing-library/react'
import { axe } from 'jest-axe'
import * as React from 'react'

import { Box } from './'

describe(`Box component`, () => {
  it('renders', async () => {
    const { container } = render(
      <Box css={{ m: 'auto', height: 100, width: 100 }}>BOX</Box>
    )
    await screen.findByText('BOX')
    expect(container).toMatchSnapshot()
  })

  it('has no programmatically detectable a11y issues', async () => {
    render(<Box />)

    const results = await axe(document.body)
    expect(results).toHaveNoViolations()
  })
})
