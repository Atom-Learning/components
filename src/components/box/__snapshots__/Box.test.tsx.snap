// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box component renders 1`] = `
.sx03kze6obnf--css {
  margin: auto;
}

.sx03kze6obnf--css {
  height: 100px;
  width: 100px;
}

<div>
  <div
    class="sx03kze sx03kze6obnf--css"
  >
    BOX
  </div>
</div>
`;
